Solutions/Marain.UserNotifications.Management.Host/Marain/UserNotifications/Management/Host/OpenApi/GenerateTemplateService.cs

```diff
@@ -0,0 +1,180 @@
+            if (userPreference == null)
+            {
+                throw new UserNotificationNotFoundException($"There is no user preference set up for this user {body.UserIds[0]} for tenant {tenant.Id}");
+            }
+
+            if (userPreference.CommunicationChannelsPerNotificationConfiguration == null)
+            {
+                throw new UserNotificationNotFoundException($"There are no communication channel set up for the user {body.UserIds[0]} for tenant {tenant.Id}");
+            }
+
+            if (!userPreference.CommunicationChannelsPerNotificationConfiguration.ContainsKey(body.NotificationType))
+            {
+                throw new UserNotificationNotFoundException($"There is no communication channel set up for the user {body.UserIds[0]} for notification type {body.NotificationType} for tenant {tenant.Id}");
+            }
+
+            List<CommunicationType>? registeredCommunicationChannels = userPreference.CommunicationChannelsPerNotificationConfiguration[body.NotificationType];
+
+            if (registeredCommunicationChannels == null || registeredCommunicationChannels.Count == 0)
+            {
+                throw new Exception($"There are no communication channel set up for the user {body.UserIds[0]} for notification type {body.NotificationType} for tenant {tenant.Id}");
+            }
+
+            // Gets the AzureBlobTemplateStore
+            INotificationTemplateStore templateStore = await this.tenantedTemplateStoreFactory.GetTemplateStoreForTenantAsync(tenant).ConfigureAwait(false);
+
+            // Get the notification template for the notification type
+            NotificationTemplate? rawNotificationTypeTemplate = await templateStore.GetAsync(body.NotificationType).ConfigureAwait(false);
+
+            if (rawNotificationTypeTemplate == null)
+            {
+                throw new UserNotificationNotFoundException($"There is no notification template set up for the user {body.UserIds[0]} for notification type {body.NotificationType} for tenant {tenant.Id}");
+            }
+
+            var communicationTypeDeliveryStatus = new Dictionary<CommunicationType, bool>();
+            EmailTemplate? emailTemplate = null;
+            SmsTemplate? smsTemplate = null;
+            WebPushTemplate? webPushTemplate = null;
+            Dictionary<string, object> existingProperties = PropertyBagHelpers.GetDictionaryFromPropertyBag(body.Properties);
+
+            foreach (CommunicationType channel in registeredCommunicationChannels)
+            {
+                switch (channel)
+                {
+                    case CommunicationType.Email:
+
+                        if (rawNotificationTypeTemplate.EmailTemplate != null && body.Properties != null)
+                        {
+                            string emailBody = await this.GenerateTemplateForField(rawNotificationTypeTemplate.EmailTemplate.Body, existingProperties);
```

Comment:
You have a mix of `await` with and without ConfigureAwait(false) in here. You might consider installing one of the threading analyzers (like https://github.com/microsoft/vs-threading/blob/master/doc/analyzers/index.md) to help catch these.

Reply:
Added this nuget to Management.Host and OpenApi projects for now just to catch errors. It did catch some naming issues but it did not catch any configureAwaits not being used.